packages/ifpack2/src/Ifpack2_Relaxation_def.hpp

```diff
@@ -143,10 +143,22 @@ namespace {
       return Teuchos::ScalarTraits<Scalar>::eps ();
     }
   };
+
+
+
 } // namespace (anonymous)
 
 namespace Ifpack2 {
 
+template<class MatrixType>
+void Relaxation<MatrixType>::updateCachedMultiVector(const Teuchos::RCP<const Tpetra::Map<local_ordinal_type,global_ordinal_type,node_type> > & map, size_t numVecs) const{
+  // Allocate a multivector if the cached one isn't perfect
+  // Note: We check for map pointer equality here since it is much cheaper than isSameAs()
+  if(cachedMV_.is_null() || &*map != &*cachedMV_->getMap() || cachedMV_->getNumVectors() !=numVecs) 
+    cachedMV_ = Teuchos::rcp(new Tpetra::MultiVector<scalar_type,local_ordinal_type,global_ordinal_type,node_type>(map, numVecs, false));
```

Comment:
Careful with the "false" bit -- not sure if it will always get zeroed out first at time of use.

Reply:
Agreed.  All of the stuff I replaced had mfh comments that said that "false" should be OK.

Reply:
I've found MueLu and Ifpack2 to be a little sloppy about zeroing out unnecessarily.

Reply:
@jhux2 I'm not sure there's a test for all the different options going on.  On the other hand, in a debug build, Tpetra::MultiVector prefills with silent NaNs when zeroOut = false.